### PR TITLE
feat: This introduces background events queue in Escoria

### PR DESCRIPTION
**Pull request by dploeger**
Currently only available in the backend. We still ned an ESC implementation to run an event in the background.

Needed this for the ink bridge though.

godot-escoria/escoria-issues#53

### CODE REVIEW

File: addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd [hunk 1]
```diff
@@ -5,11 +5,14 @@ class_name ESCEventManager
 # Emitted when the event started execution
 signal event_started(event_name)
 
+# Emitted when an event is started in a channel of the background queue
+signal background_event_started(channel_name, event_name)
+
 # Emitted when the event did finish running
-signal event_finished(event_name, return_code)
+signal event_finished(return_code, event_name)
```
Comment (StraToN) [inline addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd:12]: Changing the order of these parameters needs to be reflected in eventual yields() returned values that await this signal. There are a few in esc_room.gd (L167), esc_prompt_popup.gd (L36), escoria.gd (L342)

**Review by StraToN — CHANGES_REQUESTED** [review]

File: addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd (hunk 1 above)
Comment (dploeger) [reply addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd:12]: Actually, it doesn't. It was wrong all the time. We send (return_code, event_name), but the signal definition was turned around. :)

Comment (StraToN) [reply addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd:12]: Oh, shoot 😮 
All good then ;) Thanks!

File: addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd [hunk 2]
```diff
@@ -1,106 +1,194 @@
...
+var events_queue: Dictionary = {
+	"_front": []
+}
+
+# Currently running event in background channels
+var _running_events: Dictionary = {}
 
-# Whether the event manager is allowed to proceed with next event.
-var can_process_next_event = true
+# Wether an event can be played on a specific channel
```
Comment (StraToN) [inline addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd:34]: wHether ;)

**Review by StraToN — APPROVED** [review]

**Review by StraToN — CHANGES_REQUESTED** [review]
Currently crashing because of references to `escoria.event_manager._running_event` which was changed for `escoria.event_manager._running_events` and is now a Dictionary.

I advise using "Find in Files" in the Script editor (Ctrl+Shift+F) and search for any occurrence of "_running_event" to fix them.

**Review by StraToN — APPROVED** [review]

**Pull request merged**